File: samples/renting-java/src/main/java/renting/RentingFacade.java
package renting;

public class RentingFacade implements ForRenting {

    private final Map<BookId, Book> availableBooks;
    private final Map<BookId, Book> rentedBooks;

    public Optional<Book> rent(BookId id){
        if(!availableBooks.containsKey(id)){
            return Optional.empty();
        }

        var book = availableBooks.remove(id);
        rentedBooks.put(book);

        return Optional.of(book);
    }

}

File: samples/renting-ts/src/renting-facade.ts
import {Book, BookId} from "./book";
import {ForRenting} from "./for-renting";

class RentingFacade implements ForRenting {
    private readonly availableBooks = new Map<BookId, Book>();
    private readonly rentedBooks = new Map<BookId, Book>();

    rent(id: BookId): Book | null {
        if(!this.availableBooks.has(id)){
            return null;
        }
        const book = this.availableBooks.get(id);
        this.availableBooks.delete(id);
        this.rentedBooks.set(id, book);

        return book;
    }
}